files/pull-sandbox-image.sh

```diff
@@ -1,27 +1,5 @@
 #!/usr/bin/env bash
+set -euo pipefail
 
-### fetching sandbox image from /etc/containerd/config.toml
-sandbox_image=$(awk -F'[ ="]+' '$1 == "sandbox_image" { print $2 }' /etc/containerd/config.toml)
-region=$(echo "$sandbox_image" | cut -f4 -d ".")
-ecr_password=$(aws ecr get-login-password --region $region)
-API_RETRY_ATTEMPTS=5
-
-for attempt in `seq 0 $API_RETRY_ATTEMPTS`; do
-	rc=0
-    if [[ $attempt -gt 0 ]]; then
-        echo "Attempt $attempt of $API_RETRY_ATTEMPTS"
-    fi
-	### pull sandbox image from ecr
-	### username will always be constant i.e; AWS
-	sudo ctr --namespace k8s.io image pull $sandbox_image --user AWS:$ecr_password
-	rc=$?;
-	if [[ $rc -eq 0 ]]; then
-		break
-	fi
-	if [[ $attempt -eq $API_RETRY_ATTEMPTS ]]; then
-        exit $rc
-    fi
-    jitter=$((1 + RANDOM % 10))
-    sleep_sec="$(( $(( 5 << $((1+$attempt)) )) + $jitter))"
-    sleep $sleep_sec
-done
+sandbox_image="$(awk -F'[ ="]+' '$1 == "sandbox_image" { print $2 }' /etc/containerd/config.toml)"
+/etc/eks/containerd/pull-image.sh "${sandbox_image}"
```

Comment:
_( Discussion, I'm not sure about it )_

I can see there's a `set -e` at the very beginning and retry mechanism inside `pull-image.sh` , but there's no `set -e` in `pull-image.sh`.

I'm wondering what would happen if failed on last two lines of `pull-images.sh`, could `pull-sandbox-image.sh` detect that failure?

```
ecr_password=$(retry aws ecr get-login-password --region $region)
retry sudo ctr --namespace k8s.io image pull "${img}" --user AWS:${ecr_password}`
```

Reply:
if there's an error on retrieving the `ecr_password`, it would try the next line, but it would just retry and fail. Since the retry func exits with the proper status code for a failure, then it will be detected in `pull-sandbox-image.sh`.

Reply:
yup... I saw it, no problem for me. 💯